doc/server.py

```diff
@@ -0,0 +1,25 @@
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+from subprocess import call
+
+PORT = 8000
+URL = f"http://localhost:{PORT}"
+
+
+class AutoBrowseHTTPServer(HTTPServer):
+    def server_activate(self):
+        HTTPServer.server_activate(self)
+        print(f"Serving documentation at {URL}")
+        call(["xdg-open", URL])
```

Comment:
> The entire Python implementation can be offloaded to `http-server`, obsoleting the "[PATCH 2/2] doc: add `nix run .#docs` server" patch in this patchset

`http-server` is only a thin wrapper around the `HTTPServer` class, so I don't think that refactor gains much simplicity.

Using the `HTTPServer` class allows us to run `xdg-open` once the server has been activated; see above.

Reply:
Ah, I missed those flags. I think I was looking at the wrong `--help` reference.

I agree this is neater; I'll port it over to nixvim too!